apps/typeform/frontend/src/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,231 @@
+  expireSoon: boolean;
+}
+
+interface State {
+  workspaceId: string;
+  accessToken: string;
+  workspaces: WorkspaceOption[];
+  contentTypes: ContentType[];
+  selectedContentTypes: string[];
+  selectedFields: SelectedFields;
+  compatibleFields: CompatibleFields;
+}
+
+export class AppConfig extends React.Component<Props, State> {
+  state: State = {
+    contentTypes: [],
+    compatibleFields: {},
+    workspaces: [],
+    selectedContentTypes: [],
+    selectedFields: {},
+    workspaceId: '',
+    accessToken: (window.localStorage.getItem('token') as string) || ''
+  };
+
+  async componentWillMount() {
+    const { sdk } = this.props;
+
+    sdk.app.onConfigure(this.onAppConfigure);
+
+    const [contentTypesResponse, eisResponse, paramsResponse] = await Promise.all([
+      sdk.space.getContentTypes(),
+      sdk.space.getEditorInterfaces(),
+      sdk.app.getParameters(),
+      this.fetchWorkspaces()
+    ]);
+
+    const contentTypes = (contentTypesResponse as Hash).items as ContentType[];
+    const editorInterfaces = (eisResponse as Hash).items as EditorInterface[];
+    const compatibleFields = getCompatibleFields(contentTypes);
+    const filteredContentTypes = contentTypes.filter(ct => {
+      const fields = compatibleFields[ct.sys.id];
+      return fields && fields.length > 0;
+    });
+
+    const parameters: InstallationParameters = paramsResponse as InstallationParameters;
+
+    this.setState(
+      {
+        accessToken: get(parameters, ['accessToken'], ''),
+        workspaceId: get(parameters, ['workspaceId'], ''),
+        compatibleFields,
+        contentTypes: filteredContentTypes,
+        selectedFields: editorInterfacesToSelectedFields(editorInterfaces, sdk.ids.app)
+      },
+      () => sdk.app.setReady()
+    );
+  }
+
+  fetchWorkspaces = async () => {
+    try {
+      const response = await fetch(
+        `${process.env.LAMBDA_ENDPOINT}/workspaces/${this.state.accessToken}`
+      );
+      const result: WorkspacesResponse = await response.json();
+      this.setState({ workspaces: this.normalizeWorkspaceResponse(result) });
+    } catch (error) {
+      this.props.sdk.notifier.error(error);
+    }
+  };
+
+  normalizeWorkspaceResponse = (response: WorkspacesResponse) => {
+    return response.workspaces.items.map(workspace => ({
+      name: workspace.name,
+      id: workspace.id
+    }));
+  };
+
+  onAppConfigure = () => {
+    const { accessToken, workspaceId, contentTypes, selectedFields } = this.state;
+    const parameters = { accessToken, workspaceId };
+    const error = validateParameters(parameters);
+    const hasStaleWorkspaceIdSelected = !this.selectedWorkspaceIdIsValid();
+
+    if (error) {
+      this.props.sdk.notifier.error(error);
+      return false;
+    }
+
+    if (hasStaleWorkspaceIdSelected) {
+      this.props.sdk.notifier.error('Select a valid workspace.');
+      return false;
+    }
+
+    return {
+      parameters: { accessToken, workspaceId },
```

Comment:
We shouldn't save the access token to installation parameters. That was the reason to pivot to using OAuth.

Reply:
good catch. Was a left over from the previous implementation.